src/blame.rs

```diff
@@ -0,0 +1,539 @@
+    })
+  }
+
+  #[napi(iterator)]
+  /// Collects blame hunks by scanning file lines as an iterator
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   iterByLine(): Generator<BlameHunk>;
+  /// }
+  /// ```
+  ///
+  /// @returns Iterator of blame hunks collected by line scanning
+  /// @example
+  /// ```ts
+  /// // Using for...of loop
+  /// for (const hunk of blame.iterByLine()) {
+  ///   console.log(hunk.finalCommitId);
+  /// }
+  ///
+  /// // Using spread operator to collect all hunks
+  /// const hunks = [...blame.iterByLine()];
+  /// ```
+  pub fn iter_by_line(&self, env: Env) -> crate::Result<BlameHunksByLine> {
+    let inner = Blame {
+      inner: self.inner.clone_with_env(env)?,
+    };
+
+    Ok(BlameHunksByLine {
+      inner,
+      current_line: 1,
+      processed_hunks: HashSet::new(),
+      total_hunk_count: self.get_hunk_count() as usize,
+    })
+  }
+
+  #[napi]
+  /// Iterates through each hunk and calls the callback function
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   forEachHunk(callback: (hunk: BlameHunk, index: number) => boolean): void;
+  /// }
+  /// ```
+  ///
+  /// @example
+  /// ```ts
+  /// blame.forEachHunk((hunk, index) => {
+  ///   console.log(`Hunk ${index}: ${hunk.finalCommitId}`);
+  ///   return true; // Continue iteration
+  /// });
+  /// ```
+  ///
+  /// @param {Function} callback - Function called for each hunk
+  ///   Return true to continue iteration, false to stop
+  pub fn for_each_hunk(&self, callback: Function<(BlameHunk, u32), bool>) -> crate::Result<()> {
+    let hunk_count = self.get_hunk_count();
+
+    for idx in 0..hunk_count {
+      if let Ok(hunk) = self.get_hunk_by_index(idx) {
+        if !callback.call((hunk, idx)).unwrap_or(false) {
+          break;
+        }
+      }
+    }
+
+    Ok(())
+  }
+
+  #[napi]
+  /// Generates blame information from an in-memory buffer
+  ///
+  /// @category Blame/Methods
+  /// @signature
+  /// ```ts
+  /// class Blame {
+  ///   buffer(buffer: Buffer, bufferLen: number): Blame;
+  /// }
+  /// ```
+  ///
+  /// @example
+  /// ```ts
+  /// const buffer = Buffer.from('modified content');
+  /// const bufferBlame = blame.buffer(buffer, buffer.length);
+  /// ```
+  ///
+  /// @param {Buffer} buffer - Buffer containing file content to blame
+  /// @param {number} buffer_len - Length of the buffer in bytes
+  /// @returns A new Blame object for the buffer content
+  /// @throws If the buffer contains invalid UTF-8
+  pub fn buffer(&self, buffer: Buffer, buffer_len: u32, env: Env) -> crate::Result<Blame> {
+    let content = std::str::from_utf8(&buffer[..buffer_len as usize])?;
+
+    let blame = match &self.inner {
+      BlameInner::Repo(shared_ref) => {
+        let cloned = shared_ref.clone(env)?;
+
+        cloned.share_with(env, |git_blame| {
+          git_blame
+            .blame_buffer(content.as_bytes())
```

Comment:
```suggestion
            .blame_buffer(buffer.as_ref())
```
I'm not sure that the second parameter (`buffer_len: u32`) is needed.
`blame_buffer` requires `&[u8]` type as argument, so why not just pass the buffer instead of converting it to utf8?